init.c

```diff
@@ -560,6 +566,21 @@ EXPORT void init(struct metafile *m, int argc, char *argv[])
 
 	ll_sort(m->file_list, file_data_cmp_by_name);
 
+    /* determine the piece length based on the torrent size if
+	   it was not user specified. */
+	if (m->piece_length == 0) {
+		int i;
+		for (i = 15; i < num_piece_len_maxes &&
+			m->piece_length == 0; i++)
+			if (m->size <= piece_len_maxes[i])
+				m->piece_length = i;
+		if (m->piece_length == 0)
+			m->piece_length = i;
```

Comment:
`i = num_piece_len_maxes` in this case, so I think it would be more easily understandable if it was `m->piece_length = num_piece_len_maxes`.